crates/net/src/types/bootnode.rs

```diff
@@ -0,0 +1,33 @@
+use ethrex_core::H512;
+use std::{net::SocketAddr, num::ParseIntError, str::FromStr};
+
+#[derive(Debug, Clone)]
+pub struct BootNode {
+    pub node_id: H512,
+    pub socket_address: SocketAddr,
+}
+
+impl FromStr for BootNode {
+    type Err = ParseIntError;
+    /// Takes a str with the format "enode://<node ID>@<IP address>:<port>" and
+    /// parses it to a BootNode
+    fn from_str(input: &str) -> Result<BootNode, ParseIntError> {
+        // TODO: error handling
+        let node_id_as_bytes = decode_hex(&input[8..136])?;
+        let node_id = H512::from_slice(&node_id_as_bytes);
+        let socket_address: SocketAddr = input[137..]
+            .parse()
+            .expect("Failed to parse bootnode address and port");
+        Ok(BootNode {
+            node_id,
+            socket_address,
+        })
+    }
+}
+
+pub fn decode_hex(s: &str) -> Result<Vec<u8>, ParseIntError> {
+    (0..s.len())
+        .step_by(2)
+        .map(|i| u8::from_str_radix(&s[i..i + 2], 16))
+        .collect()
+}
```

Comment:
Can we add some tests?
For example, deserializing this EF bootnode entry: `enode://d860a01f9722d78051619d1e2351aba3f43f943f6f00718d1b9baa4101932a1f5011f16bb2b1bb35db20d6fe28fa0bf09636d26a87d31de9ec6203eeedb1f666@18.138.108.67:30303`